key.go

```diff
@@ -272,6 +274,30 @@ func (key *pKey) MarshalPKIXPublicKeyDER() (der_block []byte,
 	return ioutil.ReadAll(asAnyBio(bio))
 }
 
+// EngineLoadPrivateKey loads a private key by id
+// the id is a pkcs#11 URI https://tools.ietf.org/html/rfc7512#section-2.3
+// Engine comes from e.g.: e,err:=openssl.EngineById("pkcs11")
+func EngineLoadPrivateKey(e *Engine, id string) (PrivateKey, error) {
+	keyID := C.CString(id)
+	defer C.free(unsafe.Pointer(keyID))
+
+	if e == nil {
+		return nil, errors.New("ENGINE_load_private_key cannot be called with NULL engine")
+	}
+
+	key := C.ENGINE_load_private_key(e.e, keyID, nil, nil)
+	if key == nil {
+		return nil, errors.New("cannot load private key, ENGINE_load_private_key error")
+	}
+
+	// ref trick inspired by the work of Renato Aguiar https://github.com/renatoaguiar
```

Comment:
Question:
What does the trick do?

Reply:
guards against the freeing of the engine while the key is still in use.

Reply:
I updated the comment.